rclcpp/src/rclcpp/intra_process_manager.cpp

```diff
@@ -32,7 +32,9 @@ IntraProcessManager::~IntraProcessManager()
 {}
 
 uint64_t
-IntraProcessManager::add_publisher(rclcpp::PublisherBase::SharedPtr publisher)
+IntraProcessManager::add_publisher(
+  rclcpp::PublisherBase::SharedPtr publisher,
+  const bool is_serialized)
```

Comment:
I don't see an easy way to set a flag on the publisher base similar to the subscription base which indicates that it's a serialized publisher. Bummer.

Reply:
That being said, what happens if I take the same publisher and publish first a serialized message and then a regular `MsgT`?

Reply:
EDIT: I just browsed through the tests to see what's happening on the user side. What about we don't call a `create_publisher` function, but rather a `create_serialized_publisher` which implements the publisherbase interface (sets the `is_serialized` parameter) and only allows a call to publish with serialized messages.